app/src/main/java/de/westnordost/streetcomplete/location/LocationUtil.kt

```diff
@@ -0,0 +1,26 @@
+package de.westnordost.streetcomplete.location
+
+import android.Manifest
+import android.content.Context
+import android.content.IntentFilter
+import android.content.pm.PackageManager
+import android.location.Location
+import android.location.LocationManager
+
+import androidx.core.content.ContextCompat
+import androidx.core.location.LocationManagerCompat
+
+import de.westnordost.streetcomplete.data.osm.mapdata.LatLon
+
+fun isLocationEnabled(context: Context): Boolean =
+    hasLocationPermission(context) && LocationManagerCompat
+        .isLocationEnabled(ContextCompat.getSystemService(context, LocationManager::class.java)!!)
```

Comment:
This is what I'd rather do here, I think

```suggestion
fun isLocationEnabled(context: Context): Boolean =
    hasLocationPermission(context) && _isLocationEnabled(context)

private fun _isLocationEnabled(context: Context): Boolean {
    val lm = ContextCompat.getSystemService(context, LocationManager::class.java)!!
    return LocationManagerCompat.isLocationEnabled(lm)
}
```

Reply:
meh. Me no like

Reply:
Actually, found something nicer, one moment

Reply:
Same trick works again